### ISSUE TITLE
Issue with Node v17

### ISSUE THREAD

**Issue by leeovery** [jeansaad/chalet#23]
Hi,

I upgrade node locally yesterday and seems theres an issue with proxying (maybe) on node 17.

I downgraded to node 16 and its working fine again.

This issue is so if anyone else comes along and is seeing the Chalet error page when proxying.

FYI - the localhost connection was working fine. It was the proxying to the .test domain that was failing.

Cheers
Lee

**Comment by jeansaad** [comment]
Thanks @leeovery for reporting the issue! I can give it a try on Node v17. Would u be able to validate if it was a specific issue with the tld being `.test` or a general proxying issue?

Also was this a new machine setup or an upgrade after changing proxy settings?

**Comment by leeovery** [comment]
I actually use .box as my tld.

When my dev env broke first thing I did was to check the latest commits here but nothing seemed broken. I updated chalet yesterday but it was all fine first thing this morning. 

So I then done a clean install of chalet using .test as the tld. Same problem. 

It was at this point I remember updating brew earlier today. I checked and node was indeed updated. 

I reverted to node@16 and all was good in the world again. 

Viewing via localhost:8004 was fine. But with the .box url it was showing the chalet error page with no helpful node errors. 

The project is a Nuxt 2 / vue 2 spa. 

Hope that starts you off on the right track. Sorry I didn't get around to fixing it myself. Really appreciate you for maintaining this!!!

Lee

**Comment by leeovery** [comment]
Noticed 2nd question. I think I answered. But was not a new machine. Existing machine. Been working for ages. Updated node and chalet and it failed.

Didn't test downgrading chalet. Just downgraded node and it worked so I got back to work 😀

**Comment by jeansaad** [comment]
So the `.test` default domain is actually broken cause I thought I committed the correct file but instead I modified a generated file... 🤦🏼  Will resolve that specific problem but the default TLD is still `.localhost`.

As for v17, I did install but the actual proxy mechanism does work for me. I'm using dnsmasq + Caddy to handle the certs and https but it all forwards to chalet as an upstream proxy. Can you pinpoint if it's a cert issue or pure proxy? I find that depending on the OS, you might need a restart when changing the TLD or a browser restart.

Note: I tried with v17.0.1.

**Comment by jeansaad** [comment]
Nvm, I am able to reproduce it! I'll get back to you when I have some more information! I might also spend some time figuring out the certs too since I find it painful to get going with decent certs on all browsers cleanly.

**Comment by jeansaad** [comment]
FYI: I've addressed the `.test` TLD issue and added a `restart` command in the cli in https://github.com/jeansaad/chalet/releases/tag/v0.10.1.

**Comment by leeovery** [comment]
Nice one. 

I'll give it a whirl later. 

I use a little script with mkcert for custom SSL. Always works perfectly. I can share if it helps?

**Comment by jeansaad** [comment]
Yeah, definitely would be great if you could share! Could you share it in #14?

**Comment by leeovery** [comment]
I posted it in #14 and confirmed node 17 seems to be working fine now :)

**Comment by leeovery** [comment]
Hey.

Im sorry but I might have spoke too soon with this being ok with Node 17. Still seems broken.

**Comment by jeansaad** [comment]
@leeovery,

I tried this on v17.8.0 and it seems to work. This issue is really puzzling on how intermittent it is.
